doc/guidelines/development/docker/index.rst

```diff
@@ -4,8 +4,92 @@ Docker Guides
 
 Guides for learning how to use docker.
 
-.. toctree::
-    :glob:
-    :maxdepth: 1
+Following instructions assume you installed Docker and Visual Studio Code.
 
-    *
+First, you need to download "docker image". For example, develop-env can be installed following the instructions from here: `https://hub.docker.com/r/lsstts/develop-env/tags` 
+
+.. image:: ../../../images/install_docker_image.png
+
+You can copy the docker pull command and run it from the terminal.
+
+Now you will see image installed in your Docker app like the image below. 
+
+.. image:: ../../../images/docker_images.png
+
+There are two ways to open the container from the image you just installed. You can click the play button to start the container, or put command in terminal to run.
+
+.. image:: ../../../images/open_container_playbutton.png
+
+When opening the container using the play button, there are options you can add. You can put container name, host, volumes, and environmental variables. If you don't give a container name, it will randomly create name for you. 
+
+Similarly, when you run the container using the command, you can select options. For example:
+
+::
+
+    docker run --name lsst-dev -p 80:80 --volume ${place_where_I_save_repos}:/home/saluser/develop -it lsstts/develop-env:develop
+
+Here "--name" is the name of the container, "-p" means port, "--volume" as volumne, where you saved your repo, and "-it" as the image you are using.
+Name your repo path as "/home/saluser/develop" to reduce confusion.
+Basic idea of giving "--volume" and "-it" is that you are creating container based on the "image", and also making it to connect to your repo, which can be the git cloned repository.
+This way you can make the container and repo to communicate, testing runs in the container, update your repo by copying or movnig the files from the container, and then you can push your local repository to the origin.
+
+The containers created will show up when you click "containers" on the left side of the docker desktop as below. You can stop, or delete the containers with the icons. 
+
+.. image:: ../../../images/containers_indocker.png
+
+When you open the Visual Studio Code and go to the docker icon (whale shape) on the left column, you will see the list of images installed and containers you opened.  
```

Comment:
Maybe here you could start a new section for the vscode-docker connection. A brief description telling what this section is about may also be helpful. For example, why do we want to have this connection? What advantages it has?